### PR TITLE
Update github.com/json-iterator/go to 1.1.4

### LINKED ISSUE #66422: Sloppy vendoring of "json-iterator/go"
[json-iterator/go](https://github.com/json-iterator/go/releases) is a [semantically versioned](https://semver.org/) library yet is is [vendored by random commit](https://github.com/kubernetes/kubernetes/blob/master/Godeps/Godeps.json#L2010) instead of release tag.

Please consider vendoring by tag whenever possible.

Thanks.

### PR DESCRIPTION
**Pull request by cblecker**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
Updates this dependency to use the tagged v1.1.4. No actual go code changes.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #66422

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

### CODE REVIEW

**Comment by dims** [comment]
no code changes FTW!!!

/lgtm

File: Godeps/Godeps.json [hunk 1]
```diff
@@ -2345,8 +2345,8 @@
 		},
 		{
 			"ImportPath": "github.com/json-iterator/go",
-			"Comment": "1.1.3-22-gf2b4162afba355",
-			"Rev": "f2b4162afba35581b6d4a50d3b8f34e33c144682"
+			"Comment": "1.1.4",
+			"Rev": "ab8a2e0c74be9d3be70b3184d9acc634935ded82"
```
Comment (nikhita) [inline Godeps/Godeps.json]: I'm sure I'm probably missing something here but doesn't the commit https://github.com/json-iterator/go/commit/ab8a2e0c74be9d3be70b3184d9acc634935ded82 contain code changes?

Comment (cblecker) [reply Godeps/Godeps.json]: It changes things in the `github.com/json-iterator/go/extra` package, but that's not imported/vendored by the main `github.com/json-iterator/go` package

**Comment by cblecker** [comment]
@AishSundar As this has no changes to go code, and will ease downstream vendoring of kubernetes (improves the release), I'm going to tag this into v1.13.

/milestone v1.13
/priority critical-urgent

**Pull request merged**